src/test/mir-opt/not_equal_false.opt.InstCombine.diff

```diff
@@ -4,44 +4,35 @@
-      let mut _3: isize;                   // in scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
-      let mut _4: bool;                    // in scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-      let mut _5: isize;                   // in scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
+      let mut _2: isize;                   // in scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
+      let mut _3: isize;                   // in scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
   
       bb0: {
-          StorageLive(_2);                 // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-          _3 = discriminant(_1);           // scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
-          _2 = Eq(_3, const 0_isize);      // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-          goto -> bb4;                     // scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
+          _2 = discriminant(_1);           // scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
+          switchInt(move _2) -> [0_isize: bb3, otherwise: bb2]; // scope 0 at $DIR/not_equal_false.rs:4:17: 4:21
       }
   
       bb1: {
           _0 = const true;                 // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
-          goto -> bb3;                     // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
+          return;                          // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
       }
   
       bb2: {
-          StorageLive(_4);                 // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-          _5 = discriminant(_1);           // scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
-          _4 = Eq(_5, const 1_isize);      // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-          goto -> bb5;                     // scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
+          _3 = discriminant(_1);           // scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
+          switchInt(move _3) -> [1_isize: bb5, otherwise: bb4]; // scope 0 at $DIR/not_equal_false.rs:4:38: 4:45
       }
   
       bb3: {
-          StorageDead(_4);                 // scope 0 at $DIR/not_equal_false.rs:4:45: 4:46
-          StorageDead(_2);                 // scope 0 at $DIR/not_equal_false.rs:4:45: 4:46
-          return;                          // scope 0 at $DIR/not_equal_false.rs:5:2: 5:2
+          goto -> bb1;                     // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
       }
   
       bb4: {
-          switchInt(move _2) -> [false: bb2, otherwise: bb1]; // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
+          _0 = const false;                // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
+          return;                          // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
       }
   
       bb5: {
--         _0 = Ne(_4, const false);        // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
-+         _0 = _4;                         // scope 0 at $DIR/not_equal_false.rs:4:5: 4:46
```

Comment:
This test now does nothing during instcombine, maybe leave a comment to that regard in the source file?

Reply:
Added a more direct test for not_equal